components/General/Modalize.js

```diff
@@ -1,133 +1,160 @@
+import React, { Fragment, useState, useCallback, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React, { Fragment } from 'react';
 import { _cs, randomString } from '@togglecorp/fujs';
 
-const propTypes = {
+const controlledWrapperPropTypes = {
     disabled: PropTypes.bool,
     modal: PropTypes.element.isRequired,
     className: PropTypes.string,
     initialShowModal: PropTypes.bool,
     onClose: PropTypes.func,
+    showModal: PropTypes.bool,
+    onModalVisibilityChange: PropTypes.func,
 };
 
-const defaultProps = {
+const controlledWrapperDefaultProps = {
+    showModal: false,
     disabled: false,
     className: '',
     initialShowModal: false,
     onClose: undefined,
+    onModalVisibilityChange: undefined,
 };
 
-const modalize = (WrappedButtonComponent) => {
-    const ModalComponent = class extends React.PureComponent {
-        static propTypes = propTypes;
-
-        static defaultProps = defaultProps;
-
-        constructor(props) {
-            super(props);
-
-            this.state = {
-                showModal: props.initialShowModal,
-                wrappedButtonBCR: undefined,
-            };
-
-            this.wrappedButtonRef = React.createRef();
-            this.wrappedButtonClassName = randomString(16);
-            this.setBCRTimeout = undefined;
-            // this.wrappedButtonBCR = undefined;
-        }
-
-        componentDidMount() {
-            this.setWrappedButtonBCR();
-            window.addEventListener('resize', this.handleResize);
-            window.addEventListener('scroll', this.handleScroll, true);
-        }
-
-        componentWillUnmount() {
-            window.removeEventListener('resize', this.handleResize);
-            window.removeEventListener('scroll', this.handleScroll, true);
-        }
-
-        setWrappedButtonBCR = () => {
-            const wrappedButton = document.getElementsByClassName(this.wrappedButtonClassName)[0];
+export const controlledModalize = (WrappedButtonComponent) => {
+    const ModalComponent = (props) => {
+        const {
+            disabled,
+            modal,
+            className: classNameFromProps,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            initialShowModal,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            onClose,
+            showModal,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            onModalVisibilityChange,
+            ...otherProps
+        } = props;
+
+        const [wrappedButtonClassName] = useState(randomString(16));
```

Comment:
Use a function to set default value 
```suggestion
        const [wrappedButtonClassName] = useState(() => randomString(16));
```